### ISSUE TITLE
sf force lightning lwc test run returns exit code 0 when tests fail

### ISSUE THREAD

**Issue by jamessimone** [forcedotcom/cli#2991]
### Summary

Using latest CLI version (`@salesforce/cli/2.55.6 win32-x64 node-v20.16.0`), these two commands differ when encountering failing LWC jest tests:

* npx sfdx-lwc-jest --skipApiVersionCheck - returns exit code 1
* sf force lightning lwc test run - returns exit code 0

This improperly suppresses failures when relying on exit codes (such as in pipeline runs)

### Steps To Reproduce

```js
describe('sample failure', () => {
  it('fails', () => {
    expect(1).toBe(2);
  })
})
```

### Expected result

CLI should return exit code 1 when using ` sf force lightning lwc test run` with any test failure

### Actual result

CLI returns exit code 0

**Comment by jamessimone** [comment]
This issue should not be closed

**Comment by cristiand391** [comment]
hey, sorry for the late reply.

the command should return the same exit code that `sfdx-lwc-test` does:
https://github.com/salesforcecli/plugin-lwc-test/blob/024530c2ac27f3097f4edb075ccb1cc3b14cabf4/src/commands/force/lightning/lwc/test/run.ts#L60

but looking at the e2e test it seems that hasn't been the case:
https://github.com/salesforcecli/plugin-lwc-test/blob/024530c2ac27f3097f4edb075ccb1cc3b14cabf4/test/commands/lwc/test/run.nut.ts#L84

we'll discuss with the owners of `sfdx-lwc-test` about the current state of the lwc commands and get back.

**Comment by jamessimone** [comment]
Bump

**Comment by jamessimone** [comment]
Bump

**Comment by jamessimone** [comment]
Bump

**Comment by VivekMChawla** [comment]
@jamessimone - Thanks for keeping this bumped.

My team connected with the team that owns `sfdx-lwc-jest`

We confirmed that the behavior of `sf force lightning lwc test run` is incorrect.  The command should exit with a status code of `1` if any tests fail.

The CLI team will explore a fix for this issue, but I can't provide a timeline.

**Comment by jamessimone** [comment]
@VivekMChawla thanks for keeping us updated, it's much appreciated

**Comment by jshackell-sfdc** [comment]
This issue has been fixed in CLI version 2.62.6 (October 16, 2024).

**Comment by jamessimone** [comment]
@jshackell-sfdc thanks!
